plugin/pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -0,0 +1,69 @@
+/*
```

Comment:
@wojtek-t i *could* break it up, but my first question would be - how should we separate the changes? 

- the metadata.go changes  effect predicates.go, because now predicates.go doesnt implement the CheckService algorithm
- the changes to types.go are only meaningfull in the context of enabling matching pod metadata / node information when calcualting predicates via a factory
- the changes to plugins.go are simply machinery to enable the predicates.go metadata to be plugged in via a factory
-  also the changes to generic_scheduler are really only there for enabling factoy-ization

so its a pretty cohesive change set.   

Either way is ok, just let me know... but figured i would double-check that you think its necessary...